scheduler/util.go

```diff
@@ -334,6 +334,18 @@ func tasksUpdated(jobA, jobB *structs.Job, taskGroup string) comparison {
 			return difference("task log disabled", at.LogConfig.Disabled, bt.LogConfig.Disabled)
 		}
 
+		// Check volume mount updates
+		bVMMap := helper.SliceToMap[map[string]*structs.VolumeMount](bt.VolumeMounts,
+			func(vm *structs.VolumeMount) string {
+				return vm.Volume
+			})
+
+		for _, atvm := range at.VolumeMounts {
+			if c := volumeMountUpdated(atvm, bVMMap[atvm.Volume]); c.modified {
+				return c
+			}
+		}
```

Comment:
Because we're looping over the volume mounts for (A) but not (B), this detects changes volume mounts if both versions (A) and (B) have them, and detects if the new version (A) adds a volume mount that the existing version (B) doesn't have, but I don't think it detects if version (A) removes a volume mount that version (B) has?

See my comment on the test.

Reply:
Since we only really care about _if_ the volume mounts do not match (and order does not matter) a simple way to do this could be to stuff them in a HashSet (before and after) and ask the Sets if they are equal.

Reply:
@shoenig Done!